chiselFrontend/src/main/scala/chisel3/core/BiConnect.scala

```diff
@@ -89,9 +102,40 @@ object BiConnect {
         case _ => recordConnect(sourceInfo, connectCompileOptions, left_r, right_r, context_mod)
       }
 
+      // Handle Records connected to DontCare (change to NotStrict)
+      case (left_r: Record, DontCare) =>
+        left_r.compileOptions match {
+          case ExplicitCompileOptions.NotStrict =>
+            left.bulkConnect(right)(sourceInfo, ExplicitCompileOptions.NotStrict)
+          case _ =>
+            // For each field in left, descend with right
+            for ((field, left_sub) <- left_r.elements) {
+              try {
+                connect(sourceInfo, connectCompileOptions, left_sub, right, context_mod)
+              } catch {
+                case BiConnectException(message) => throw BiConnectException(s".$field$message")
+              }
+            }
+        }
+      case (DontCare, right_r: Record) =>
```

Comment:
There seems to be a lot of code duplication here, which might be better off in a separate function that gets called, between this and the Record-Record connection? Similarly with Vec, though to a lesser degree.

This isn't a huge issue, though.

Reply:
I tried, but couldn't come up with anything that didn't duplicate the existing logic without adding additional code.

Reply:
Yeah, that's fine, I couldn't come with anything obvious in the 2 minutes or so I thought about it.